tests/test_core.py

```diff
@@ -19,175 +20,176 @@ def req():
         yield req
 
 
+def test_download_defaults(monkeypatch, mocker):
+    download_mock = mocker.MagicMock()
+    monkeypatch.setattr(core, 'download', download_mock)
+    core.download()
+    assert download_mock.call_count == len(core.SUPPORTED_DOMAINS)
+
+
+# TODO: test unrecognized arguments, invalid formats and out of choices values
+
 def test_download_one(monkeypatch, mocker):
-    _download_mock = mocker.MagicMock()
-    monkeypatch.setattr(core, '_download', _download_mock)
-    fake_args = Namespace(section='refseq', domain='bacteria', uri=core.NCBI_URI,
-                          output='/tmp/fake', file_format='genbank', assembly_level='all',
-                          genus='', species_taxid=None, taxid=None, human_readable=False,
-                          parallel=1)
-    core.download(fake_args)
-    _download_mock.assert_called_with('refseq', 'bacteria', core.NCBI_URI, '/tmp/fake',
-                                      'genbank', 'all', '', None, None, False, 1)
+    download_mock = mocker.MagicMock()
+    monkeypatch.setattr(core, 'download', download_mock)
+    kwargs = {'domain': 'bacteria', 'output': '/tmp/fake'}
+    core.download(**kwargs)
+    download_mock.assert_called_with(**kwargs)
 
 
 def test_download_all(monkeypatch, mocker):
-    _download_mock = mocker.MagicMock()
-    monkeypatch.setattr(core, '_download', _download_mock)
-    fake_args = Namespace(section='refseq', domain='all', uri=core.NCBI_URI,
-                          output='/tmp/fake', file_format='genbank', assembly_level='all',
-                          genus='', species_taxid=None, taxid=None, human_readable=False,
-                          parallel=1)
-    core.download(fake_args)
-    assert _download_mock.call_count == len(core.SUPPORTED_DOMAINS)
+    download_mock = mocker.MagicMock()
+    monkeypatch.setattr(core, 'download', download_mock)
+    core.download(domain='all', output='/tmp/fake')
+    assert download_mock.call_count == len(core.SUPPORTED_DOMAINS)
```

Comment:
Indeed, in that code version, the test was not passing. It is now passing again, after being updated to the changes of `core.py`.